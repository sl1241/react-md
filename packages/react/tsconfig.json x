{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
