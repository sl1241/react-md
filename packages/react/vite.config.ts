import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
