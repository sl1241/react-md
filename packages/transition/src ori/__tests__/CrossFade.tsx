import type { ReactElement } from "react";
import { useState } from "react";
import { act, fireEvent, render } from "@testing-library/react";

import type { CrossFadeProps } from "../CrossFade";
import { CrossFade } from "../CrossFade";

interface TestProps
  extends Omit<CrossFadeProps<HTMLElement>, "transitionIn" | "children"> {
  defaultTransitionIn?: boolean;
}

function Test({
  defaultTransitionIn = false,
  ...props
}: TestProps): ReactElement {
  const [transitionIn, setTransitionIn] = useState(defaultTransitionIn);
  return (
    <>
      <button type="button" onClick={() => setTransitionIn((p) => !p)}>
        Toggle
      </button>
      <CrossFade {...props} appear={false} transitionIn={transitionIn}>
        <div data-testid="element">This is some content.</div>
      </CrossFade>
    </>
  );
}

describe("CrossFade", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllTimers();
  });

  it("should default to using appear transitions", () => {
    const { container, getByTestId, rerender } = render(
      <CrossFade>
        <div data-testid="element">This is some content.</div>
      </CrossFade>
    );
    const getElement = (): HTMLElement => getByTestId("element");

    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    rerender(
      <CrossFade key="new-transition">
        <div data-testid="element">This is some new content.</div>
      </CrossFade>
    );
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();
  });

  it("should transition correctly when not appearing", () => {
    const { container, getByRole, getByTestId } = render(<Test />);
    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("");
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    // no transition for exit
    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("");
    expect(container).toMatchSnapshot();

    // shouldn't really be any timers
    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("");
    expect(container).toMatchSnapshot();
  });

  it("should allow for temporary CSS Transition behavior", () => {
    const { container, getByRole, getByTestId } = render(<Test temporary />);
    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    // no transition for exit
    fireEvent.click(toggle);
    expect(getElement).toThrow();
  });

  it("should correctly merge the className prop", () => {
    const { container, getByRole, getByTestId } = render(
      <Test className="custom-class" />
    );
    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("custom-class");
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(getElement()).toHaveClass("custom-class");
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("custom-class");
    expect(container).toMatchSnapshot();

    // no transition for exit
    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("custom-class");
    expect(container).toMatchSnapshot();

    // shouldn't really be any timers
    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(getElement().className).toBe("custom-class");
    expect(container).toMatchSnapshot();
  });

  it("should correctly call the transition callbacks", () => {
    const onEnter = jest.fn();
    const onEntering = jest.fn();
    const onEntered = jest.fn();
    const onExit = jest.fn();
    const onExiting = jest.fn();
    const onExited = jest.fn();
    const props: TestProps = {
      onEnter,
      onEntering,
      onEntered,
      onExit,
      onExiting,
      onExited,
    };

    const { getByRole } = render(<Test {...props} />);
    const toggle = getByRole("button");

    expect(onEnter).not.toBeCalled();
    expect(onEntering).not.toBeCalled();
    expect(onEntered).not.toBeCalled();
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).not.toBeCalled();
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    act(() => {
      jest.runAllTimers();
    });
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).toBeCalledTimes(1);
  });
});
