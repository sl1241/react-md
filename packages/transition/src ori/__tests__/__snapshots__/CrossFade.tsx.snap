// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrossFade should allow for temporary CSS Transition behavior 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`CrossFade should allow for temporary CSS Transition behavior 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-cross-fade rmd-cross-fade--active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should allow for temporary CSS Transition behavior 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should correctly merge the className prop 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should correctly merge the className prop 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class rmd-cross-fade rmd-cross-fade--active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should correctly merge the className prop 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should correctly merge the className prop 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should correctly merge the className prop 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should default to using appear transitions 1`] = `
<div>
  <div
    class="rmd-cross-fade rmd-cross-fade--active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should default to using appear transitions 2`] = `
<div>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should default to using appear transitions 3`] = `
<div>
  <div
    class="rmd-cross-fade rmd-cross-fade--active"
    data-testid="element"
  >
    This is some new content.
  </div>
</div>
`;

exports[`CrossFade should default to using appear transitions 4`] = `
<div>
  <div
    data-testid="element"
  >
    This is some new content.
  </div>
</div>
`;

exports[`CrossFade should transition correctly when not appearing 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should transition correctly when not appearing 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-cross-fade rmd-cross-fade--active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should transition correctly when not appearing 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should transition correctly when not appearing 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`CrossFade should transition correctly when not appearing 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;
